examples/campaign_management/add_complete_campaigns_using_batch_job.rb

```diff
@@ -0,0 +1,370 @@
+  response = batch_job_service_client.mutate_batch_job(
+    customer_id: customer_id,
+    operation: operation
+  )
+
+  batch_job_resource_name = response.result.resource_name
+  puts "Created a batch job with resource name: '#{batch_job_resource_name}'"
+
+  batch_job_resource_name
+end
+
+# Adds all batch job operations to the batch job. As this is the first time
+# for this batch job, pass null as a sequence token. The response will contain
+# the next sequence token that you can use to upload more operations in the
+# future.
+def add_all_batch_job_operations(
+  client,
+  batch_job_service_client,
+  customer_id,
+  batch_job_resource_name)
+  response = batch_job_service_client.add_batch_job_operations(
+    resource_name: batch_job_resource_name,
+    mutate_operations: build_all_operations(client, customer_id),
+  )
+  puts "#{response.total_operations} mutate operations have been added so far."
+
+  # You can use this next sequence token for calling
+  # add_all_batch_job_operations() next time
+  puts "Next sequence token for adding next operations is " \
+    "'#{response.next_sequence_token}'"
+end
+
+# Requests the API to run the batch job for executing all uploaded batch job
+# operations.
+def run_batch_job(batch_job_service_client, batch_job_resource_name)
+  operation_response = batch_job_service_client.run_batch_job(
+    resource_name: batch_job_resource_name,
+  )
+  puts "Batch job with resource name '#{batch_job_resource_name}' " \
+    "has been executed."
+  operation_response
+end
+
+# Polls the server until the batch job execution finishes by setting the initial
+# poll delay time and the total time to wait before time-out.
+def poll_batch_job(operation_response)
+  operation_response.wait_until_done!
+end
+
+# Prints all the results from running the batch job.
+def fetch_and_print_results(batch_job_service_client, batch_job_resource_name)
+  puts "Batch job with resource name '#{batch_job_resource_name}' has " \
+    "finished. Now, printing its results..." \
+
+  # Gets all the results from running batch job and print their information.
+  batch_job_results = batch_job_service_client.list_batch_job_results(
+    resource_name: batch_job_resource_name,
+    page_size: PAGE_SIZE,
+  )
+  batch_job_results.each do |result|
+    puts "Batch job ##{result.operation_index} has a status " \
+      "#{result.status ? result.status.message : 'N/A'} and response of type " \
+      "#{result.mutate_operation_response ? result.mutate_operation_response.response : 'N/A'}"
+  end
+end
+
+# Builds all operations for creating a complete campaign and return an array of
+# their corresponding mutate operations.
+def build_all_operations(client, customer_id)
+  mutate_operations = []
+
+  # Creates a new campaign budget operation and add it to the array of mutate
+  # operations.
+  campaign_budget_operation = build_campaign_budget_operation(client, customer_id)
+  mop = client.operation.mutate
```

Comment:
"mop" is a word that has nothing to do with what you're doing. I think you were going for something like "mutate_op", but maybe spell it out so it doesn't sound silly to native speakers :)

Reply:
Makes sense :) Changed to mutate_op